docs/source/release_notes.rst

```diff
@@ -23,6 +23,7 @@ Release Notes
         * Updated ``add_results`` in ``AutoMLAlgorithm`` to take in entire pipeline results dictionary from ``AutoMLSearch`` :pr:`1891`
         * Updated ``ClassImbalanceDataCheck`` to look for severe class imbalance scenarios :pr:`1905`
         * Deleted the ``explain_prediction`` function :pr:`1915`
+        * Deleted data check code in ``AutoMLSearch`` :pr:`1935`
```

Comment:
I think this is also a breaking change. We should also mention this means we're deleting the `data_check_results` property and that `data_checks` is not a valid `search` argument anymore.

Reply:
Agreed! Let's add a sentence or two in the breaking changes about how to handle this, too. Something like "Please run the data checks before AutoML like so: `results = DefaultDataChecks().validate(X_train, y_train)`"